### PR TITLE
chore: Fork rs-ucan into noosphere-ucan and noosphere-ucan-key-support

### PR DESCRIPTION
**Pull request by jsantell**
Bump internal version of ed25519-zebra to 4.

### CODE REVIEW

**Comment by jsantell** [comment]
Rebased unsigned-varint collision with https://github.com/subconsciousnetwork/noosphere/pull/825

File: rust/noosphere-ucan/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
+![API Stability: Alpha](https://img.shields.io/badge/API%20Stability-Alpha-red)
+
+# Noosphere Ucan
```
Comment (cdata) [inline rust/noosphere-ucan/README.md:3]: Can you leave a comment here regarding the `v0.10.0 -> v1.0.0` major breaking changes as a rationale for the fork? Something like: "This project was forked from the official `rs-ucan` implementation, pending the finalization and broad adoption of the v1.0.0 spec (which represents a major breaking change from v0.10.0)."

**Review by cdata — APPROVED** [review]
Approved, assuming slight touch-ups. Thanks for doing this!

File: rust/noosphere-storage/Cargo.toml [hunk 2]
```diff
@@ -19,13 +19,13 @@ readme = "README.md"
...
 anyhow = { workspace = true }
-async-trait = "~0.1"
+async-trait = { workspace = true }
 async-stream = { workspace = true }
 tokio-stream = { workspace = true }
 cid = { workspace = true }
 noosphere-common = { version = "0.1.2", path = "../noosphere-common" }
 tracing = "~0.1"
-ucan = { workspace = true }
+noosphere-ucan = { version = "0.4.0", path = "../noosphere-ucan" }
```
[the remark this answers is not in the document]
Comment (jsantell) [reply rust/noosphere-storage/Cargo.toml]: Seems to work outside of the `-dev` scenarios (#629), as it looks for a workspace dependency with name e.g. `noosphere-ns-dev`
```toml
# Doesn't work
noosphere-ns-dev = { workspace = true, package = "noosphere-ns" }
# OK
noosphere-ns-dev = { path = "../noosphere-ns", package = "noosphere-ns" }
```

**Comment by jsantell** [comment]
Patching the latest nightly lints as well https://github.com/subconsciousnetwork/noosphere/actions/runs/7978793781/job/21784685442?pr=815

**Pull request merged**